mongodbatlas/fw_resource_mongodbatlas_project.go

```diff
@@ -0,0 +1,862 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"reflect"
+	"sort"
+	"time"
+
+	"go.mongodb.org/atlas-sdk/v20230201002/admin"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/setplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	expanders "github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/framework/expanders"
+)
+
+const (
+	errorProjectCreate  = "error creating Project: %s"
+	errorProjectRead    = "error getting project(%s): %s"
+	errorProjectDelete  = "error deleting project (%s): %s"
+	errorProjectSetting = "error setting `%s` for project (%s): %s"
+	errorProjectUpdate  = "error updating project (%s): %s"
+)
+
+var _ resource.Resource = &ProjectRS{}
+var _ resource.ResourceWithImportState = &ProjectRS{}
```

Comment:
not sure if I see value in these statements, checking of correct interface implementation will be done as soon as resource is configured in the provider

Reply:
Could you please elaborate `as soon as resource is configured in the provider`?

Reply:
When the resource is added into the provider, it will have to comply with the resource.Resource interface: 
`return []func() resource.Resource{
		NewProjectRS,
}`


I do however see that for the ImportState interface there would be no compilation errors if something is incorrectly defined, so line 42 does have value.
To add, hashicorp example uses this convention: https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/main/internal/provider/example_resource.go#L22 so we can follow with it.